### DISCUSSION TITLE
[Feature] Separate folder for generated artifacts (thumbs etc) from the immich UPLOAD_LOCATION folder

### DISCUSSION THREAD

**Discussion by dgtal1** [immich-app/immich#8884] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Would be great if it was possible to define separate folder location for immich upload files and separate folder for generated files like thubnails etc. Currently all of it sits in the `./library` folder (by default), which is not ideal.

I like the approach Photoprism took - it separates 'storage' from the 'original' photos to index. 

Basically the scenario I want to achieve is to use immich to index my existing external photo library stored on my Synology NAS. I do not want to reupload everything again to immich. Instead I just want to index and use the upload feature maybe in the future. 
I store my original, source photos on my NAS. I made a volume in docker-compose for it and added the entire folder as external library. 
However from the performance point of view - I want immich to work fast, so I want its thumbnails to be stored on my SSD not HDD with the original photos. When I define the UPLOAD path and map it to a folder on my SSD connected to my NAS - it's all good until I will want to upload photos with immich. Then they will be uploaded to my SSD, but I want them to be uploaded to a dedicated folder on my RAID HDD on my Synology. Ideally it would be the next  subfolder of the main folder mapped as external library to immich.

If it was possible to define the UPLOAD folder separately I could achieve what I need - have thumbnails on my SSD and photos uploaded by immich to my RAID HDD. 

Currently it's not possible since everything is kept together.

I believe it is not a big deal to split it, so I'm hoping someone can really do it easily in the next builds :)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This isn't really necessary to do on an app level, since it can be done with docker mountpoints: https://immich.app/docs/guides/custom-locations

**Reply by dgtal1** [reply]
Yes, it indeed looks liek this should be possible to solve with docker! :) Thanks! I'm still new to immich and docker-compose. I used Photoprism shortly and I learned the very basics with it but apparently there's still a lot more to learn :)

**Reply by miq-codes** [reply]
@dgtal1 Do you mind sharing your docker file and .env. I was trying to keep uploaded files separately and all immich media separately.

**Reply by dgtal1** [reply]
> @dgtal1 Do you mind sharing your docker file and .env. I was trying to keep uploaded files separately and all immich media separately.

I'll check in the first moment when I log onto my server (and if I don't forget :) ) but it's quite straightforward - the guide / link above explain it. You leave the folders in the container intact and just point the external folders to whatever you like - these don't have to be nested as in the inside, immich folders

**Reply by pandaboy6621** [reply]
I am also curious as to how you set those env variables, Im not too happy about how the immich app upload renames a the IMG filename and then isn't categorized by date like a typical camera roll folder would be.

**Reply by bo0tzz** [reply]
@pandaboy6621 you're looking for https://immich.app/docs/administration/storage-template

**Reply by zhen-huan-hu** [reply]
> This isn't really necessary to do on an app level, since it can be done with docker mountpoints: https://immich.app/docs/guides/custom-locations

Maybe it's nick-picking but using this method, not only the docker compose file and env file become quite clustered, Immich would also create empty folders under the `UPLOAD_LOCATION` location for `profile`, `thumbs`, `encoded-video` and `backups`, even though the actual files will be stored elsewhere.

If only `upload` and `library` are in a separate directory by default, all of these can be avoided.
